cli/pcluster/config/param_types.py

```diff
@@ -0,0 +1,1202 @@
+                    related_section = related_section_type(self.related_section_map, label)
+
+                    for param_key, param_map in self.related_section_map.get("params").items():
+                        cfn_converter = param_map.get("cfn", None)
+                        if cfn_converter:
+
+                            param_type = param_map.get("type", Param)
+                            cfn_value = get_cfn_param(cfn_params, cfn_converter).split(",")[index]
+                            param = param_type(
+                                self.section_key,
+                                self.section_label,
+                                param_key,
+                                param_map,
+                                self.pcluster_config,
+                                cfn_value=cfn_value,
+                            )
+                            related_section.add_param(param)
+
+                            if param.value != param_map.get("default", None):
+                                configured_params = True
+
+                    if configured_params:
+                        self.pcluster_config.add_section(related_section)
+
+        self.value = ",".join(labels) if labels else None
+
+    def to_file(self, config_parser):
+        """Convert the param value into a list of sections in the config_parser and initialize them."""
+        sections = {}
+        if self.value:
+            for section_label in self.value.split(","):
+                sections.update(self.pcluster_config.get_section(self.related_section_key, section_label.strip()))
+
+        if sections:
+            config_section_name = _get_file_section_name(self.section_key, self.section_label)
+            try:
+                # add parent section, if not present
+                config_parser.add_section(config_section_name)
+            except DuplicateSectionError:
+                LOGGER.debug("Section '[%s]' is already present in the file.", config_section_name)
+                pass
+            # add "*_settings = *" to the parent section
+            config_parser.add_section(config_section_name)
+
+            # create sections
+            for _, section in sections:
+                section.to_file(config_parser)
+
+    def to_cfn(self):
+        """Convert a list of sections to multiple CFN params."""
+        sections = OrderedDict({})
+        if self.value:
+            for section_label in self.value.split(","):
+                section = self.pcluster_config.get_section(self.related_section_key, section_label.strip())
+                sections.update({section_label: section})
+
+        max_number_of_ebs_volumes = 5
+
+        cfn_params = {}
+        number_of_ebs_sections = len(sections)
+        for param_key, param_map in self.related_section_map.get("params").items():
+            if number_of_ebs_sections == 0 and param_key == "shared_dir":
```

Comment:
if number_of_ebs_sections == 0 why even looping through the params?

Reply:
Because we need to populate all the EBS related CFN parameters with default values.